src/status_im/contexts/wallet/create_account/new_keypair/check_your_backup/view.cljs

```diff
@@ -0,0 +1,127 @@
+(ns status-im.contexts.wallet.create-account.new-keypair.check-your-backup.view
+  (:require
+    [quo.core :as quo]
+    [react-native.core :as rn]
+    [reagent.core :as reagent]
+    [status-im.contexts.wallet.common.temp :as temp]
+    [status-im.contexts.wallet.create-account.new-keypair.check-your-backup.style :as style]
+    [utils.i18n :as i18n]
+    [utils.re-frame :as rf]))
+
+(def secret-words-count 12)
+
+(defn- random-selection
+  []
+  (->> (range secret-words-count)
+       (shuffle)
+       (take 4)
+       vec))
+
+(defn- random-words-with-string
+  [array given-string]
+  (let [random-words (->> array
+                          (remove #(= % given-string))
+                          (shuffle)
+                          (take 3))
+        result       (conj random-words given-string)]
+    (->> result
+         (shuffle)
+         vec)))
+
+(defn- cheat-warning
+  []
+  (let [{:keys [customization-color]} (rf/sub [:profile/profile])]
+    [rn/view
+     [quo/drawer-top {:title (i18n/label :t/do-not-cheat)}]
+     [quo/text
+      {:style style/cheat-description}
+      (i18n/label :t/do-not-cheat-description)]
+     [quo/bottom-actions
+      {:button-one-label (i18n/label :t/see-recovery-phrase-again)
+       :button-one-props {:customization-color customization-color
+                          :on-press            (fn []
+                                                 (rf/dispatch [:hide-bottom-sheet])
+                                                 (rf/dispatch [:navigate-back {:revealed? true}]))}}]]))
```

Comment:
@OmarBasem - Is there any reason behind adding a param `revealed?`?

Reply:
@smohamedjavid yes, navigating to the previous screen with `revealed?` true will show the secret phrase immediately

Reply:
I believe we are not using it in the previous screen. We need to get the screen param from the sub `:get-screen-params`.

Reply:
Thanks @smohamedjavid, you are right. I have removed it cause it is actually not needed. The secret phrase will already be revealed as we are just navigating back so the state is saved.